src/su/context.rs

```diff
@@ -23,6 +27,19 @@ pub(crate) struct SuContext {
     pub(crate) process: Process,
 }
 
+/// check that a shell is not restricted / exists in /etc/shells
+fn is_restricted(shell: &Path) -> bool {
+    if let Some(pattern) = shell.as_os_str().to_str() {
+        if let Ok(contents) = fs::read_to_string(VALID_LOGIN_SHELLS_LIST) {
+            if contents.contains(pattern) {
```

Comment:
`/etc/shells` can contain comments. this will consider commented out shells as non-restricted shells

Reply:
Nice catch - I'll check whether the shell corresponds to a full line in this file

Reply:
Fixed in [ba72489](https://github.com/memorysafety/sudo-rs/pull/590/commits/ba72489170125c479fd97e3d00c04dd898ea5e73)